atoma-proxy/src/server/handlers/chat_completions.rs

```diff
@@ -791,283 +828,962 @@ impl RequestModel for RequestModelChatCompletions {
+                refusal: Option<String>,
+                /// The tool calls generated by the model, such as function calls.
+                #[serde(default, skip_serializing_if = "Vec::is_empty")]
+                tool_calls: Vec<tools::ToolCall>,
+                /// Data about a previous audio response from the model.
+                #[serde(default, skip_serializing_if = "Option::is_none")]
+                audio: Option<audio::Audio>,
+            },
+            /// The role of the messages author, in this case tool.
+            Tool {
+                /// The contents of the message.
+                #[serde(default, skip_serializing_if = "Option::is_none")]
+                content: Option<message_content::MessageContent>,
+                /// Tool call that this message is responding to.
+                #[serde(default, skip_serializing_if = "String::is_empty")]
+                tool_call_id: String,
+            },
+        }
+    }
 
-    /// The chat completion message.
-    pub message: ChatCompletionMessage,
+    pub mod message_content {
+        use serde_json::Value;
+
+        use super::{Deserialize, Deserializer, Serialize, ToSchema};
+
+        /// Represents the content of a message.
+        ///
+        /// This is used to represent the content of a message in the chat completion request.
+        /// It can be either a text or an array of content parts.
+        #[derive(Debug, PartialEq, Eq, Serialize, ToSchema)]
+        #[serde(untagged)]
+        pub enum MessageContent {
+            /// The text contents of the message.
+            #[serde(rename(serialize = "text", deserialize = "text"))]
+            Text(String),
+            /// An array of content parts with a defined type, each can be of type text or image_url when passing in images.
+            /// You can pass multiple images by adding multiple image_url content parts. Image input is only supported when using the gpt-4o model.
+            #[serde(rename(serialize = "array", deserialize = "array"))]
+            Array(Vec<MessageContentPart>),
+        }
 
-    /// The reason the chat completion was finished.
-    #[schema(example = "stop")]
-    pub finish_reason: Option<String>,
+        /// Represents a part of a message content.
+        ///
+        /// This is used to represent the content of a message in the chat completion request.
+        /// It can be either a text or an image.
+        #[derive(Debug, PartialEq, Eq, Serialize, Deserialize, ToSchema)]
+        #[serde(untagged)]
+        pub enum MessageContentPart {
+            #[serde(rename(serialize = "text", deserialize = "text"))]
+            Text {
+                /// The type of the content part.
+                #[serde(rename(serialize = "type", deserialize = "type"))]
+                r#type: String,
+                /// The text content.
+                text: String,
+            },
+            #[serde(rename(serialize = "image", deserialize = "image"))]
```

Comment:
can we add the audio as well?

Reply:
Not in this PR, we will need to add support later